### PR TITLE
[backport] Remove host-side branch on `F.accuracy` with `ignore_label`

### PR DESCRIPTION
**Pull request by niboshi**
Backport of #8364

### CODE REVIEW

**Comment by niboshi** [comment]
[automatic post] Jenkins, test this please.

**Comment by emcastillo** [comment]
Seems like Jenkins failures are related

File: chainer/functions/evaluation/accuracy.py [hunk 1]
```diff
@@ -45,10 +46,16 @@ def forward(self, inputs):
...
+                # Avoid warning of `divide by zero`
+                if total == 0:
+                    return xp.asarray(0.0, dtype=y.dtype),
+                else:
+                    return xp.asarray(float(count) / total, dtype=y.dtype),
             else:
-                return xp.asarray(float(count) / total, dtype=y.dtype),
+                return xp.where(total == 0,
+                                xp.asarray(0.0, dtype=y.dtype),
+                                xp.asarray(count / total, dtype=y.dtype)),
```
Comment (emcastillo) [inline chainer/functions/evaluation/accuracy.py]: py2 division problem?

Comment (niboshi) [reply chainer/functions/evaluation/accuracy.py]: Thank you for pointing out. Fixed in 564f38825acc05453cdbf35e7fa7fe24578de293.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Doesnt seem an easy fix   😞

**Pull request closed without merging**